File: dx-api/app/consts/user_grade.go
package consts

// User grade values.
const (
	UserGradeFree     = "free"
	UserGradeMonth    = "month"
	UserGradeSeason   = "season"
	UserGradeYear     = "year"
	UserGradeLifetime = "lifetime"
)

// UserGradeLabels maps each grade to its Chinese label.
var UserGradeLabels = map[string]string{
	UserGradeFree:     "免费会员",
	UserGradeMonth:    "月度会员",
	UserGradeSeason:   "季度会员",
	UserGradeYear:     "年度会员",
	UserGradeLifetime: "终身会员",
}

// UserGradePrices maps each grade to its price in CNY.
var UserGradePrices = map[string]int{
	UserGradeFree:     0,
	UserGradeMonth:    39,
	UserGradeSeason:   99,
	UserGradeYear:     309,
	UserGradeLifetime: 1999,
}

// UserGradeMonths maps each grade to the number of months it grants.
// A value of 0 means not applicable (free has no months, lifetime is permanent).
var UserGradeMonths = map[string]int{
	UserGradeFree:     0,
	UserGradeMonth:    1,
	UserGradeSeason:   3,
	UserGradeYear:     12,
	UserGradeLifetime: 0,
}

File: dx-web/src/consts/user-grade.ts
export const USER_GRADES = {
  FREE: "free",
  MONTH: "month",
  SEASON: "season",
  YEAR: "year",
  LIFETIME: "lifetime",
} as const;

export type UserGrade = (typeof USER_GRADES)[keyof typeof USER_GRADES];

export const USER_GRADE_PRICES: Record<UserGrade, number> = {
  free: 0,
  month: 39,
  season: 99,
  year: 309,
  lifetime: 1999,
};

export const USER_GRADE_LABELS: Record<UserGrade, string> = {
  free: "免费会员",
  month: "月度会员",
  season: "季度会员",
  year: "年度会员",
  lifetime: "终身会员",
};

/** Number of months each grade adds when redeemed */
export const USER_GRADE_MONTHS: Record<UserGrade, number | null> = {
  free: null,
  month: 1,
  season: 3,
  year: 12,
  lifetime: null,
};
